File: algorithm/203. Remove Linked List Elements/203.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        a=ListNode(0)
        a.next=head
        tem=a
        while tem and tem.next:
            while tem.next and tem.next.val==val:
                tem.next=tem.next.next
            tem=tem.next
        return a.next

File: algorithm/203. Remove Linked List Elements/203.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode a= ListNode(0);
        a.next=head;
        ListNode* tem=&a;
        while(tem && tem->next)
        {
            while(tem->next && tem->next->val==val)
            {
                tem->next=tem->next->next;
            }
            tem=tem->next;
        }
        return a.next;
    }
};
